### ISSUE TITLE
Add ??? flag to variable-name rule

### ISSUE THREAD

**Issue by unional** [palantir/tslint#1328]
### Bug Report
- **TSLint version**: 3.10.2
- **TypeScript version**: 1.9.0-dev.20160614-1.0
- **Running TSLint via**: VSCode
#### TypeScript code being linted

``` ts
export interface Foo {

}

// [tslint] variable name must be in camelcase or uppercase
export const Foo = { 
  someFunction() {}
}
```

Is there a better construct without hitting this warning or getting into class + static method?
how can I disable this rule only for this use case (maybe adding a `bypass-overload` flag)?

**Comment by JKillian** [comment]
You could add the `allow-pascal-case` option to allow this format of name in general.

**Comment by unional** [comment]
Yes, but that's what I don't want to. 
This is really a type + utility construct. And being a type it should be uppercase. 
The utility part is really a namespace idea, not a variable.

**Comment by wOvalle** [comment]
I'm facing this issue, I don't want to allow all my variables to be pascal-case but the exported ones (mostly react functional components.

**Comment by ajafff** [comment]
@wovalle 
I'd like to suggest one of my custom rules: https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/naming-convention.md#examples

Example config for your use case:

```json
"naming-convention": [
    true,
    {"type": "variable", "format": "camelCase"},
    {"type": "variable", "modifiers": "export", "format": "PascalCase", "final": true}
]

**Comment by JoshuaKGoldberg** [comment]
Duplicate of #522
